### ISSUE TITLE
repo not ready for ibundle with basis_seq_num 0

### ISSUE THREAD

**Issue by sean-anderson-seco** [drmikehenry/git-ibundle#1]
Hi, I tried following the instructions in the readme, but they don't seem to work.

```bash
git --version
git ibundle --version
mkdir a
touch a/file
git -C a init
git -C a add file
git -C a commit -m init
git -C a ibundle create ../a.ibundle
git init --bare b
git -C b ibundle fetch -v ../a.ibundle
```

Output is:

```
git version 2.25.1
git-ibundle 0.2.0
Initialized empty Git repository in /path/to/a
[master (root-commit) 1776a82] init
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 file
wrote '../a.ibundle': seq_num 1, added 2, removed 0, moved 0, unchanged 
Initialized empty Git repository in /path/to/b/
read '../a.ibundle': seq_num 1, added 2, removed 0, moved 0, unchanged 0
repo is missing 1 prerequisites listed in ibundle
  1776a82cbb026d98b25514115c763d5693274e56 'init'
repo is missing 2 orefs for basis_seq_num 0
  1776a82cbb026d98b25514115c763d5693274e56 'HEAD'
  1776a82cbb026d98b25514115c763d5693274e56 'refs/heads/master'
error: repo not ready for ibundle with basis_seq_num 0
```

I tried using `--standalone`, synchronizing with a regular bundle first, and using `git clone` to create `b`, but nothing seems to work.

**Comment by drmikehenry** [comment]
Thanks for your very clear report.  I hadn't realized that the minimum Git version I'd specified is too old.  git-ibundle uses `git bundle create --stdin` to create bundle files.  In my early experiments, I'd thought this usage was working on older Git because the `--stdin` switch is recognized without error,  but proper support for `--stdin` requires Git v2.31.0 or later.

On my system (with Git version 2.34.1), the following works with your example `a` repository to create a temporary bundle file:

```console
$ echo HEAD | git -C a bundle create -q ../temp.bundle --stdin && perl -pe 'if (/^$/) { exit; }' < temp.bundle
# v2 git bundle
7bcc9402103ab6fc63dc3cfb2747a9871f811e60 HEAD
```

With Git version 2.25.1, the switch appears to be accepted but Git acts as though `HEAD` were not being requested on stdin:

```console
$ echo HEAD | git -C a bundle create -q ../temp.bundle --stdin && perl -pe 'if (/^$/) { exit; }' < temp.bundle
fatal: Refusing to create empty bundle.
```

git-ibundle treats the `Refusing to create empty bundle` error case specially, as an empty bundle file can be a valid result, so this case looks like a reasonable response to git-ibundle (unlike, say, providing an invalid switch like `--bogus` to Git that git-ibundle would treat as an error).

The below commit added support for `git bundle create --stdin`:
<https://github.com/git/git/commit/8b48981987956d7cb10c5d7cddbf08d5837bc753>

> "git bundle" learns "--stdin" option to read its refs from the standard input.  Also, it now does not lose refs whey they point at the same object.

This was released in Git version 2.31.0.

I'll update the git-ibundle documentation to specify Git v2.31.0 as the minimum required version.  Thanks again, and sorry for the incorrect version specification.

**Comment by sean-anderson-seco** [comment]
Thanks. Upgrading git fixes my issue.
